cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs

```diff
@@ -0,0 +1,156 @@
+
+pub struct OperatorWorker {
+    worker_id: u32,
+    func_name: String,
+    input: Option<OperatorWorkerInput>,
+}
+
+impl OperatorWorker {
+    pub fn new() -> Self {
+        OperatorWorker {
+            worker_id: 0,
+            func_name: "operator".to_string(),
+            input: None,
+        }
+    }
+}
+
+enum OperatorKind {
+    Add,
+    Sub,
+}
+
+struct OperatorWorkerInput {
+    op: OperatorKind,
+    cipher_op1: String,
+    cipher_op2: String,
+}
+
+impl Worker for OperatorWorker {
+    fn function_name(&self) -> &str {
+        self.func_name.as_str()
+    }
+
+    fn set_id(&mut self, worker_id: u32) {
+        self.worker_id = worker_id;
+    }
+
+    fn id(&self) -> u32 {
+        self.worker_id
+    }
+
+    fn prepare_input(&mut self, dynamic_input: Option<String>) -> Result<()> {
+        let msg = dynamic_input.ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        // `args` should be "op,op_num,op1,op2,op3,..."
+        // now `op` may be 'add' or 'sub'
+        // number is parsed with big endian
+
+        let splited = Vec::from_iter(msg.split("something").map(String::from));
+
+        let op = splited.get(0).unwrap();
+
+        match op.as_str() {
+            "add" => {
+                let op_num = splited.get(1).unwrap().parse::<u64>().unwrap();
+                if op_num != 2 {
+                    return Err(Error::from(ErrorKind::InvalidInputError));
+                }
+
+                let cipher_op1 = splited.get(2).unwrap();
+                let cipher_op2 = splited.get(3).unwrap();
+
+                self.input = Some(OperatorWorkerInput { 
+                    op: OperatorKind::Add,
+                    cipher_op1: cipher_op1.to_string(),
+                    cipher_op2: cipher_op2.to_string() });
+            },
+            "sub" => {
+                let op_num = splited.get(1).unwrap().parse::<u64>().unwrap();
+                if op_num != 2 {
+                    return Err(Error::from(ErrorKind::InvalidInputError));
+                }
+                let cipher_op1 = splited.get(2).unwrap();
+                let cipher_op2 = splited.get(3).unwrap();
+
+                self.input = Some(OperatorWorkerInput { op:OperatorKind::Sub,
+                    cipher_op1: cipher_op1.to_string(),
+                    cipher_op2: cipher_op2.to_string() });
+            },
+            _ => {
+                return Err(Error::from(ErrorKind::InvalidInputError));
+            }
+        }
+
+        Ok(())
+
+    }
+
+    fn execute(&mut self, _context: WorkerContext) -> Result<String> {
+        let input = self
+            .input
+            .take()
+            .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        let key_pair = register_func::get_key_pair();
+        let s1 = vec![];
+        let s2 = vec![];
+        let op1 = eigen_crypto::ec::suite_b::ecies::decrypt(key_pair, &input.cipher_op1.as_bytes(), &s1, &s2).unwrap();
```

Comment:
Decrypt AES Key by ECIES, the decrypt cipher oprand by AES key.

Reply:
OK. I've added these in the latest commitment.